daemon.go

```diff
@@ -146,3 +147,41 @@ func (d *Daemon) listen() {
 		go d.handleConn(c)
 	}
 }
+
+func clearUnixSockets(maddr ma.Multiaddr) error {
+	for _, subMaddr := range ma.Split(maddr) {
```

Comment:
Use `ma.SplitFirst` here as the unix part would be first here.

Reply:
Currently, it's only expected that there is one unix socket in a multiaddr. But there is no real conceptual reason for that to be the case, and it might change in the future, without someone remembering to update it here. Should we still use SplitFirst? 

I also looked at ForEach to loop over all of the Components, but ForEach doesn't have a way to propagate errors

Reply:
I think it's unlikely to have multi-unix multi-addrs (what a mouthful :), as such addresses wouldn't have a meaningful representation (unix multiaddrs consume the rest of the multiaddr representation).
Using `SplitFirst` will result in simpler code, so it is recommended in this case.

Reply:
Updated